util/testutil/addcat/addcat.go

```diff
@@ -0,0 +1,39 @@
+package testutil
+
+import (
+	"bytes"
+	"strings"
+
+	core "github.com/ipfs/go-ipfs/core"
+	coreunix "github.com/ipfs/go-ipfs/core/coreunix"
+	unixfs "github.com/ipfs/go-ipfs/shell/unixfs"
+	errors "github.com/ipfs/go-ipfs/util/debugerror"
+)
+
+func AddCat(adder *core.IpfsNode, catter *core.IpfsNode, data string) error {
```

Comment:
On Fri, May 01, 2015 at 04:56:06AM -0700, Juan Batiz-Benet wrote:
> > +func AddCat(adder *core.IpfsNode, catter *core.IpfsNode, data string) error {
> 
> why a `string` and not `[]byte`? 

From the commit message for 87001b78 (util/testutil/addcat/addcat:
Factor out add/cat/verify helper, 2015-04-26):

  I used a string (instead of []byte) for AddCat's data, because we
  don't need to mutate the value, and Go's strings are just immutable
  byte arrays…

Also related are my comments from 3af39ac3 (cmd/ipfs/init:
bytes.NewBufferString -> strings.NewReader, 2015-04-25):

  From the docs for strings.NewReader [1]:

  It is similar to bytes.NewBufferString but more efficient and
  read-only.

  [1]: https://golang.org/pkg/strings/#NewReader

Basically, I agree with Joshua Poehls [2]:

  Ask not when to use string but rather, when to use []byte. Always
  start with string and switch to []byte when justified.

> it's a test so it doesn't matter much, but this likely should be
> `[]byte`. we want to deal with large blocks of data as `[]byte`.

Do we need the mutability in many places?  If so, I agree that it's
probably better to use []byte everywhere.  But otherwise it seems
better to use the immutable type as widely as possible and only
perform a copy-conversion to a mutable type when we need the
mutability.

[2]: http://joshua.poehls.me/2014/04/go-101-string-or-byte-slice/